src/components/structuredefSettings/StructuredefSettings.tsx

```diff
@@ -70,7 +75,11 @@ const StructuredefSettings: React.FC<StructuredefSettingsProps> = ({
       editExperimental(structureDefinitonToEdit, experimental);
       editTitle(structureDefinitonToEdit, title);
       editCopyright(structureDefinitonToEdit, copyright);
-      dispatch(getStructureDefSuccess(structureDefinitonToEdit));
+      if (type === "resource") {
+        dispatch(getStructureDefSuccess(structureDefinition));
+      } else if (type === "extension") {
+        dispatch(updateStructureDefExtension(structureDefinition));
```

Comment:
I think the action names you use in this case should be almost the same, as their usage is alike.

It seems like the action 'getStructureDefSuccess' may not be well named as it is used to update the store located structure def

Reply:
I replaced getStructureDefSuccess by updateStructureDefProfile